processor/metricstransformprocessor/metrics_transform_processor_helper.go

```diff
@@ -0,0 +1,372 @@
+// Copyright 2020 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metricstransformprocessor
+
+import (
+	"fmt"
+	"math"
+	"math/rand"
+
+	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
+	"github.com/golang/protobuf/ptypes/timestamp"
+)
+
+// timeseriesGroupByLabelValues is a data structure for grouping timeseries that will be aggregated
+type timeseriesGroupByLabelValues struct {
+	// key is a composite of the label values as a single string
+	// keyToTimeseriesMap groups timeseries by the label values
+	keyToTimeseriesMap map[string][]*metricspb.TimeSeries
+	// keyToLabelValuesMap maps the keys to the actual label values objects
+	keyToLabelValuesMap map[string][]*metricspb.LabelValue
+}
+
+// update updates the metric content based on operations indicated in transform.
+func (mtp *metricsTransformProcessor) update(metric *metricspb.Metric, transform Transform) {
+	// metric name update
+	if transform.NewName != "" {
+		metric.MetricDescriptor.Name = transform.NewName
+	}
+
+	for _, op := range transform.Operations {
+		switch op.Action {
+		case UpdateLabel:
+			mtp.updateLabelOp(metric, op)
+		case AggregateLabels, AggregateLabelValues:
+			mtp.aggregateOp(metric, op)
+		}
+	}
+}
+
+// updateLabelOp updates labels and label values in metric based on given operation
+func (mtp *metricsTransformProcessor) updateLabelOp(metric *metricspb.Metric, op Operation) {
+	for idx, label := range metric.MetricDescriptor.LabelKeys {
+		if label.Key != op.Label {
+			continue
+		}
+		// label key update
+		if op.NewLabel != "" {
+			label.Key = op.NewLabel
+		}
+		// label value update
+		labelValuesMapping := op.ValueActionsMapping
+		for _, timeseries := range metric.Timeseries {
+			newValue, ok := labelValuesMapping[timeseries.LabelValues[idx].Value]
```

Comment:
There are uses cases where the existing value of the label won't be known and it needs to be updated to the new value. Is there a way to support updating the value by only knowing the label key and not the original value?

Ex. Label K/V {'environment': 'prodUction'}/ {`environment`: 'PRODUCTION`}..., usually collectors are run within a certain environment so it is known that there is a typo in the environment value but every application had a particular way to set it so when navigating to a ui it owuld be best to blanket update all values of environment to one value 'production' regardless of what their previous value was.

Reply:
I might need to take a closer look to understand this fully. This might need more discussions to see how to update this operation entirely.

Reply:
It does sound reasonable to have the option to just update all datapoints to a given label value (similar to what add_label is going to do in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/441)

Maybe that can be the default behaviour if `Value` (current value) is not supplied. Or maybe we should force users to make explicit by having another Action (might be overkill?).

Definitely leave that for a separate PR though.